### ISSUE TITLE
MPICH will not build with MPE

### ISSUE THREAD

**Issue by Samg381** [pmodels/mpich#6013]
When configuring MPICH with the following command:
`./configure --with-mpilibs="[-L$PREFIX/lib] -lmpich -lmpl" --with-mpiinc="-I$PREFIX/include" --with-mpicc=$PREFIX/bin/mpicc`

The configuration completes, however with the following error:
`configure: WARNING: unrecognized options: --with-mpilibs, --with-mpiinc, --with-mpicc`

How do I configure MPICH with MPE enabled?

**Comment by roblatham00** [comment]
Wow, MPE is old school.  I hope it still works but we have not touched that code since 2015.

We don't build MPICH with MPE enabled.   Instead, think of MPE as an external library that uses MPI

Please try https://github.com/pmodels/mpe 

For example, I was able to configure MPE against my MPICH installation like this

    configure CC=mpicc FC=mpifort MPI_F77=mpifort

**Comment by Samg381** [comment]
I will be testing this shortly and reporting back. Appreciate the responses thus far- I will close if I have success.

**Comment by Samg381** [comment]
I was able to build MPE with the help of this thread - thank you @roblatham00 
**Unfortunately, I failed to realize MPE is no longer compatible with Python which is a requirement for my project.**

Could someone recommend a logging library that would allow me to instrument a Python application? I am specifically trying to capture communication events between threads. 

Any recommendations would be appreciated. Feel free to relocate this issue.

**Comment by hzhou** [comment]
> I am specifically trying to capture communication events between threads.

Could you elaborate? Are you sending self messages between threads?

**Comment by Samg381** [comment]
> Could you elaborate?

Certainly. I have a simple Python program with a few `comm.Bcast` and `comm.Gather` calls that I would like to have logged. I am interested in how long each thread takes to complete the task. **I am trying to accomplish this without having to touch the source code**, as there may come a time where I need to capture the same data from a program whose source code is not available. Ideally this would be exported to a text / csv file.

**Comment by hzhou** [comment]
@dalcinl Do you know/recommend any easy way to do profiling in Python?

**Comment by dalcinl** [comment]
@Samg381 I guess you are attempting to use MPE to do MPI profiling while using mpi4py? 

MPE could still be used with mpi4py, as long as you manage to build MPE as a shared library. I had a lot of code to do that automatically in mpi4py, but given that MPE has lagged behind, I eventually removed some stuff.
Again, MPE can still be used easily within mpi4py, as long as you have MPE built as a shared library. Basically, all you would need to do is:

```
import mpi4py
mpi4py.profile("/path/to/libmpe.so")
from mpi4py import MPI
...
```

Even easier, you could just do:
```
export LD_PRELOAD=/path/to/libmpe.so
mpiexec -n 5 python -m mpi4py.bench helloworld
```
and you should get the MPE log files (IIRC, you can use a environ var to define the log file names).

However, there is still an issue: mpi4py development (git branch `master`) has moved to use the large-count MPI-4 APIs exclusively. MPE has absolutely zero support for the new MPI-4 stuff. So your only option is to use mpi4py from the latest tarball release, or from the git `maint` branch.

@hzhou @roblatham00 Unfortunately, there is no free and easy solution that I know to profile MPI-based codes in Python. It is a pity that MPE had lagged behind, but I understand that resources are not infinite, and you guys cannot keep burning taxpayer dollars on everything.
